sherpa/models/model.py

```diff
@@ -543,6 +543,7 @@ def __getitem__(self, filter):
     def startup(self, cache):
         self._queue = ['']
         self._cache = {}
+        self._use_caching = cache
         if int(self.cache) > 0:
```

Comment:
So, it looks like we have two "variables" to control the cache: the `cache` argument to `startup`, which says "do we cache or not", and the `cache` attribute of the model object, which defines how many items are in the cache (by defining the length of the `_queue` list). Given that we now require python 3.5 at a minimum, we could look at using the [lru_cache](https://docs.python.org/3.5/library/functools.html#functools.lru_cache) decorator rather than writing our own, but that's outside the scope of this PR.